account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,192 @@
+import { type ConnectionConfig, type SmartAccountSigner } from "@aa-sdk/core";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import type { Address } from "abitype";
+import {
+  type Hex,
+  type SignableMessage,
+  type TypedData,
+  type TypedDataDefinition,
+  hashMessage,
+  hashTypedData,
+} from "viem";
+import { TurnkeyClient, type TSignedRequest } from "@turnkey/http";
+import type { SignRawMessageMode, User } from "./client/types.js";
+import { buildStampedSignatureRequestBody } from "./client/base.js";
+import { AlchemySignerClient } from "./client/alchemy.js";
+
+interface ApiKeySignerClientParams {
+  orgId: string;
+  apiKey: {
+    publicKey: string;
+    privateKey: string;
+  };
+  connection: ConnectionConfig;
+}
+
+/**
+ * ApiKeySignerClient is a client that uses an API key to sign messages.
+ */
+class ApiKeySignerClient {
+  protected turnkeyClient: TurnkeyClient;
```

Comment:
I'm open to other opinions, but marking these `protected` implies that this class is intended to be subclassed, and I have a strong aversion to subclassing classes which are not abstract. I would recommend (almost) never subclassing in general, but in particular I think subclassing non-abstract classes is a giant pile of worms. Prefer marking this as `private` or making the intent of how this class should be subclassed clear.

Happy to go into more detail about this: subclassing is one of my biggest pet peeves.

Reply:
thanks, will keep that in mind & change it if we end up keeping this. i'm pretty new to abstract classes tbh. (most of the typescript i've written has followed more functional patterns rather than OOP.)

i'm about to put up another pr that handles things a bit differently though.